scalapact-sbtplugin/src/main/scala/com/itv/scalapact/plugin/verifier/Verifier.scala

```diff
@@ -37,39 +37,47 @@ object Verifier {
 
           case \/-((b, p, c)) =>
             val address = b + "/pacts/provider/" + p + "/consumer/" + c + "/latest"
-            println(s"Attempting to fetch pact from pact broker at: $address".white.bold)
+            readPact(address)
+        }
+      }.collect {
+        case Some(s) => s
+      }
 
-            Http(address).header("Accept", "application/json").asString match {
-              case r: HttpResponse[String] if r.is2xx =>
-                val pact = ScalaPactReader.jsonStringToPact(r.body).toOption
+      val versionedPacts : List[Pact] = pactVerifySettings.versionedConsumerNames.map { consumer =>
+        val (consumerName, consumerVersion) = consumer
+        val details = for {
+          consumerName <- Helpers.urlEncode(consumerName)
+          validatedAddress <- PactBrokerAddressValidation.checkPactBrokerAddress(pactVerifySettings.pactBrokerAddress)
+          providerName <- Helpers.urlEncode(pactVerifySettings.providerName)
+        } yield (validatedAddress, providerName, consumerName, consumerVersion)
 
-                if(pact.isEmpty) {
-                  println("Could not convert good response to Pact:\n" + r.body)
-                  pact
-                } else pact
+        details match {
+          case -\/(l) =>
+            println(l.red)
+            None
 
-              case _ =>
-                println(s"Failed to load consumer pact from: $address".red)
-                None
-            }
+          case \/-((b, p, c, v)) =>
+            val address = b + "/pacts/provider/" + p + "/consumer/" + c + "/version/" + v
+            readPact(address)
         }
       }.collect {
         case Some(s) => s
       }
+
+      latestPacts ++ versionedPacts
```

Comment:
We talked about this but I couldn't remember the outcome: Did you want to validate BOTH (consumerA, latest) and (consumerA, 123) if it appears in both lists?